docs/important/relevant_functionalities_de.md

```diff
@@ -0,0 +1,1250 @@
+
+`create-sa.sh`
+
+```bash
+#!/bin/bash
+set -o errexit
+set -o nounset
+set -o pipefail
+doguName="${1}"
+{
+  #1) Erstellen der Zugangsdaten
+  USER="${doguName}-$(doguctl random)"
+  PASSWORD=$(doguctl random)
+
+  #2) Legt einen Account für unsere Software mit den Informationen an
+  # ... Implementiere die Dogu-spezifische Logik hier
+} >/dev/null 2>&1 # unterdrücke sonstige Ausgaben auf stdout, da nur Zugangsdaten oder Fehler erwartet werden
+
+  #3) Gib Zugangsdaten zurück. Der Client schreibt die Zugangsdaten verschlüsselt in den Dogu-spezifischen Konfigurationsbereich des Verbrauchers.
+echo "username: ${USER}"
+echo "password: ${PASSWORD}"
+```
+
+In Schritt 1 werden Zugangsdaten zufällig generiert. Der Name des Konsumenten-Dogus kann in die Zugangsdaten eingearbeitet werden, allerdings ist dies keine Pflicht.
+
+In Schritt 2 muss ein Account mit den generierten Informationen in der Dogu-Software erstellt werden. 
+
+In Schritt 3 werden die Zugangsdaten mit dem Format `echo "<registrykey>: <registryvalue>"` ausgegeben. `registrykey` und `registryvalue` müssen durch einen Doppelpunkt und einem Leerzeichen getrennt sein. Die gesamte Zeile muss durch ein Newline beendet werden.
+
+Diese Zugangsdaten werden vom verarbeitenden Client automatisch eingelesen und in den Dogu-spezifischen Konfigurationsbereich des Verbrauchers verschlüsselt abgelegt.
+Nach dem obigen Beispiel würde dem Konsumenten-Dogu zwei Registry-Einträge einrichten, die das jeweilige Zugangsdatum als Wert enthalten:  
+
+- `/config/<konsument>/sa-<produzent>/username`
+- `/config/<konsument>/sa-<produzent>/passwort` 
+
+Das Verbaucher-Dogu kann diese nun z. B. durch `doguctl config -e sa-<produzentdogu>/username` auslesen und entschlüsseln.
+
+#### Service Account löschen
+
+Das `service-account-remove`-Skript löscht einen bestehenden Service Account eines Konsumenten-Dogus.
+
+Es wird immer ausgeführt, wenn das Konsumenten-Dogu deinstalliert wird. Während des Aufrufs erhält das `service-account-remove`-Skript jedenfalls den Namen des Konsumenten-Dogus als erstes Argument. Da hier keine Ausgaben des Skriptes verarbeitet werden, können auch Fehlermeldungen auf `stdout` ausgegeben werden.
+
+Ein `service-account-remove`-Skript besteht aus zwei Schritten:
+
+1. Identifier von dem Konsumenten identifizieren 
+2. Löschen der Daten und des Service Accounts des Konsumenten
+
+Ein Beispiel für das `service-account-remove`-Skript könnte folgendermaßen aussehen:
+
+`remove-sa.sh`
+
+```bash
+#!/bin/bash
+set -o errexit
+set -o nounset
+set -o pipefail
+
+#1) Identifier von dem Konsumenten identifizieren
+SERVICE="${1}"
+if [ X"${SERVICE}" = X"" ]; then
+  echo "usage remove-sa.sh servicename"
+  exit 1
+fi
+
+#2) Löschen des Service Accounts des Verbrauchers
+# ... Implementiere die Dogu-spezifische Logik hier
+```
+
+In Schritt 1 wird der Name des Konsumenten-Dogus ausgelesen (z. B. "redmine"), i. d. R. um den Service Account des Dogus zu identifizieren.
```

Comment:
```suggestion
In Schritt 1 wird der Name des Konsumenten-Dogus ausgelesen (z. B. "redmine"), in der Regel um den Service Account des Dogus zu identifizieren.
```

Reply:
I think the whole phrase can be removed.